### PR TITLE
Add trial price to car trial validation tool

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Display trial price in car trial validation tool

Some refactoring
- Safer types
- Reuse response type between API route and UI components
- Catch every error in API route, before we had nothing if error happened before fetch (ex. missing env variable)

## Justify why they are needed

Dealership request - more info is always better, I guess

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
* **#4018** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4018?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4011** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4011?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/features/carDealership/CarBuyerValidationPage.tsx [hunk 1]
```diff
@@ -135,76 +136,67 @@ export const CarBuyerValidationPage = (props: Props) => {
...
+        </ResultRow>
+        <DisplayRow title="Personnummer:" displayValue={props.ssn} color="signalGreenText" />
+        <DisplayRow
+          title="Registreringsnummer:"
+          displayValue={props.registrationNumber}
+          color="signalGreenText"
+        />
+        <DisplayRow
+          title="Månadspris första 60 dagarna:"
+          displayValue={`${props.trialPrice.amount} ${props.trialPrice.currencyCode} / må`}
```
Comment (guilhermespopolin) [inline apps/store/src/features/carDealership/CarBuyerValidationPage.tsx:169]: There's a `monthlyPrice` formatter. Maybe we can use it here for keeping consistency

**Review by guilhermespopolin — APPROVED** [review]
Well done. Just left a small comment.

File: apps/store/src/features/carDealership/CarBuyerValidationPage.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/features/carDealership/CarBuyerValidationPage.tsx:169]: This is what I started with, but bringing full i18n infrastructure just for it seems like an overkill.  Perhaps some time later.  This page is similar to debug tools in not requiring any i18n

**Comment by alebedev** [comment]
### Merge activity

* **Mar 6, 8:32 AM EST**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4018).

**Pull request merged**